src/Debugger/Debuggee.php

```diff
@@ -0,0 +1,285 @@
+     *            controller service.
+     *      @type string $project Project the debuggee is associated with. Use
+     *            the project number when registering a Google Cloud Platform
+     *            project.
+     *      @type string $uniquifier Debuggee uniquifier within the project. Any
+     *            string that identifies the application within the project can
+     *            be used. Including environment and version or build IDs is
+     *            recommended.
+     *      @type string $description Human readable description of the
+     *            debuggee. Including a human-readable project name, environment
+     *            name and version information is recommended.
+     *      @type string $isInactive If set to true, indicates that the debuggee
+     *            is considered as inactive by the Controller service.
+     *      @type string $agentVersion Version ID of the agent release. The
+     *            version ID is structured as following:
+     *            domain/type/vmajor.minor (for example
+     *            google.com/gcp-java/v1.1).
+     *      @type string $status Human readable message to be displayed to the
+     *            user about this debuggee. Absence of this field indicates no
+     *            status. The message can be either informational or an error
+     *            status.
+     *      @type ExtendedSourceContext[] $extSourceContexts References to the locations and
+     *            revisions of the source code used in the deployed application.
+     */
+    public function __construct(ConnectionInterface $connection, array $info = [])
+    {
+        $this->connection = $connection;
+
+        $info += [
+            'id' => null,
+            'isInactive' => false,
+            'agentVersion' => DebuggerClient::DEFAULT_AGENT_VERSION,
+            'status' => null,
+            'extSourceContexts' => [],
+            'uniquifier' => null,
+            'description' => null
+        ];
+
+        $this->id = $info['id'];
+        $this->project = $info['project'];
+        $this->uniquifier = $info['uniquifier'];
+        $this->description = $info['description'];
+        $this->status = $info['status'];
+        $this->agentVersion = $info['agentVersion'];
+        $this->isInactive = $info['isInactive'];
+        $this->extSourceContexts = $info['extSourceContexts'];
+    }
+
+    /**
+     * Return the debuggee identifier.
+     *
+     * Example:
+     * ```
+     * echo $debuggee->id();
+     * ```
+     *
+     * @return string
+     */
+    public function id()
+    {
+        return $this->id;
+    }
+
+    /**
+     * Register this debuggee with the Stackdriver backend.
+     *
+     * Example:
+     * ```
+     * $debuggee->register();
+     * echo $debuggee->id();
+     * ```
+     *
+     * @param array $args
+     * @return bool
+     */
+    public function register(array $args = [])
```

Comment:
The typical convention for methods which trigger network requests is for this to be named `$options`